uro98/assignment4/CountIslands.java

```diff
@@ -0,0 +1,57 @@
+
+public class CountIslands {
+
+  /**
+   * Returns the number of islands in a given map.
+   * @param rowCount  the number of rows in the map
+   * @param colCount  the number of columns in the map
+   * @param map       the boolean map where true is land and false is water
+   * @return the number of islands in integer
+   */
+  public static int countIslands(int rowCount, int colCount, boolean[][] map) {
```

Comment:
rowCount and colCount are not really needed - dimensions can be extracted from the map itself. If you want to keep them, it would be good to check if values passed as rowCount and colCount match actual dimensions of the map to avoid bugs.

Reply:
I'd like to keep them because the assignment doc specified that they are part of the function input. I'll add dimension checks.